SonarQube.Client/Api/V6_30/GetPropertiesRequest.cs

```diff
@@ -61,8 +62,28 @@ public override async Task<SonarQubeProperty[]> InvokeAsync(HttpClient httpClien
             return result.Value;
         }
 
-        private SonarQubeProperty ToProperty(PropertyResponse arg) =>
-            new SonarQubeProperty(arg.Key, arg.Value);
+        private IEnumerable<SonarQubeProperty> ToProperties(PropertyResponse arg)
+        {
+            if (arg.FieldValues != null)
+            {
+                for (int i = 0; i < arg.FieldValues.Length; i++)
+                {
+                    var fieldValue = arg.FieldValues[i];
+                    foreach (var item in fieldValue)
+                    {
+                        yield return new SonarQubeProperty($"{arg.Key}.{i + 1}.{item.Key}", item.Value);
```

Comment:
One day, I'm going to profile this use of `yield`, and when the performance and memory allocations turn out to be much worse than just building and returning a list (not to mention making debugging harder), I'm going to write a rule to say "don't use yield like this" ;-)
Mind you, it could be difficult to separate out the false positives... ;-)

Reply:
The code is not performance critical, it should not matter much if you use yield or return array/list... :)

Reply:
True - it's just horrible to debug through ;-)